packages/core/src/lib/types.mts

```diff
@@ -0,0 +1,12 @@
+export type F<A extends any[], B> = (...args: A) => B;
+export type UnaryF<T, R> = F<[T], R>;
+export type Action<T> = UnaryF<T, void>;
+
+export type Activatable<T> = {
+  active: T;
+  setActive: Action<T>;
+};
+
+export type HasSeveralOrdered<T> = {
+  items: readonly T[];
+};
```

Comment:
Let's inline those types as much as possible (even if it leads to some duplication; not "DRY")